riptide-core/src/test/java/org/zalando/riptide/TryWithTest.java

```diff
@@ -0,0 +1,187 @@
+package org.zalando.riptide;
+
+/*
+ * ⁣​
+ * Riptide: Core
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.zalando.riptide.TryWith.tryWith;
+
+import java.io.Closeable;
+import java.nio.charset.CharacterCodingException;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+public class TryWithTest {
```

Comment:
A general rule of thumb: stubbing (`when`) and verification (`verify`) should be seen as mutually exclusive. 99% of tests should only ever use one xor the other. Very, very rarely you need to use both for the same mock.